src/main/java/com/atlassian/bitbucket/jenkins/internal/jenkins/auth/TrustedJenkinsAuthorizer.java

```diff
@@ -0,0 +1,44 @@
+package com.atlassian.bitbucket.jenkins.internal.jenkins.auth;
+
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.auth.TrustedUnderlyingSystemAuthorizerFilter;
+import com.atlassian.bitbucket.jenkins.internal.applink.oauth.serviceprovider.exception.NoSuchUserException;
+import hudson.model.User;
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import static java.util.Objects.requireNonNull;
+
+public class TrustedJenkinsAuthorizer implements TrustedUnderlyingSystemAuthorizerFilter {
+
+    private static final Logger log = Logger.getLogger(TrustedJenkinsAuthorizer.class.getName());
+
+    @Override
+    public void authorize(String userName, HttpServletRequest request, HttpServletResponse response,
+                          FilterChain filterChain) throws IOException, ServletException, NoSuchUserException {
+        requireNonNull(userName, "userName");
+        User u = getUser(userName);
+        if (u != null) {
+            try (ACLContext ignored = createACLContext(u)) {
+                log.info("Successfully logged in as user " + userName);
+                filterChain.doFilter(request, response);
+            }
+        } else {
+            throw new NoSuchUserException();
```

Comment:
is there any benefit to logging the username? (warn maybe?)

Reply:
I have passed in the user name. Although the current caller knows it and does not use the message.